cumulusci/core/flows.py

```diff
@@ -72,26 +77,46 @@ def _init_flow(self):
         self.logger.info('---------------------------------------')
 
         self.logger.info('')
-        self._init_org()
+        if not self.nested:
+            self._init_org()
         for line in self._render_config():
             self.logger.info(line)
 
+    def _check_infinite_flows(self, tasks, flows=None):
+        if flows == None:
+            flows = []
+        for task in tasks.values():
+            if 'flow' in task:
+                flow = task['flow']
+                if flow in flows:
+                    raise ConfigError('Infinite flows detected')
+                flows.append(flow)
+                flow_config = self.project_config.get_flow(flow)
+                self._check_infinite_flows(flow_config.tasks, flows)
+
     def _get_tasks(self):
+        if not self.nested:
+            self._check_infinite_flows(self.flow_config.tasks)
         tasks = []
         for step_num, config in list(self.flow_config.tasks.items()):
-            if config['task'] == 'None':
+            if 'flow' in config and 'task' in config:
+                raise ConfigError('"flow" and "task" in same config item')
+            if (('flow' in config and config['flow'] == 'None') or
+                ('task' in config and config['task'] == 'None')):
                 continue
```

Comment:
we support no-op tasks?

Reply:
Hmm good catch, I suppose this should raise a config exception?

Reply:
On second thought this keeps the functionality we had before (see deleted line 82) so I'll leave it like this. I think the idea is to no-op when overriding a task in another flow, like using a deploy+test flow but skipping tests.

Reply:
that makes a ton of sense. can you add a comment that this is to allow overrides in yaml?